### PR TITLE
Created tests for wrangler worker

### LINKED ISSUE #13: Add `tests` for `worker` using `jest`
## Description
This issue is related to the PR created for #1.

We need to write tests for our Cloudflare `workers` using `jest`. The `worker/index.test.js` was automatically generated along with `worker/index.js` using a script.

### Additional Information
 The test file is outdated since the `worker/index.js` file has undergone significant changes since it was initially created.

### PR DESCRIPTION
**Pull request by devils2ndself**
* updated package.json with new npm scripts and dependencies
  * refactored raw response and login form
  * fixed and rewrote worker.fetch unit test

Resolves #13

### CODE REVIEW

**Comment by devils2ndself** [comment]
I got a bit lost in POST request code, and I genuinely don't think we could write a unit test for such a complex action. @SerpentBytes could you help me with your thoughts and suggestions?

File: package.json
Comment (SerpentBytes) [inline package.json:9]: We shouldn't replace an existing script, as we might still need it. Maybe give your script a different name?

File: worker/index.test.js
Comment (SerpentBytes) [inline worker/index.test.js:5]: What kind of worker are we testing? "Worker" isn't clear enough. I think.

**Review by SerpentBytes — CHANGES_REQUESTED** [review]
Thank you for your PR.

I tested your PR, and it seems to be working.  

Here's a sample of a test written using `jest` to test a worker: https://github.com/pmrotule/cloudflare-worker-jest-test/blob/main/src/handler.spec.ts

I am interested in other devs feedback on this, so let's wait.

**Comment by SerpentBytes** [comment]
Thanks for your PR.

Here's a sample of a test written using `jest` for Cloudflare worker: 
https://github.com/pmrotule/cloudflare-worker-jest-test/blob/main/src/handler.spec.ts

Given that the project is still in its infancy, I am unsure if the tests have to be too extensive. 

Let's wait for other collaborators' feedback on this.

File: package.json
Comment (devils2ndself) [reply package.json:9]: I thought about it, but ideally there should be 1 command for running all tests, which is jest is great for, considering that in the future, react tests could be written using jest. 

For now, I can bring that command back and rename it to something along the way of `pwa:test`. Let me know what you think

Comment (SerpentBytes) [reply package.json:9]: Fair enough.

**Comment by devils2ndself** [comment]
Changed the description, and added a couple of tests for checking the headers

**Review by SerpentBytes — APPROVED** [review]
LGTM, but I will wait for feedback from other devs, before merging.

**Comment by SerpentBytes** [comment]
@devils2ndself Could you please update the `CONTRIBUTING.md` with information on how to test the code?

**Comment by devils2ndself** [comment]
Done! If there's nothing else regarding the test, would it be possible to merge it by end of tomorrow? I wanted to use this one as my pull request for Release 0.3

**Review by saminarp — APPROVED** [review]
It looks good to me!

**Review by SerpentBytes — APPROVED** [review]

File: worker/index.test.js
Comment (batunpc) [inline worker/index.test.js:2]: rawHtmlResponse is not being used

File: worker/index.test.js
Comment (batunpc) [inline worker/index.test.js:3]: > When I save this file in my editor eslint automatically insterts `""` instead `''`

please use quotes `" "` instead `''` to keep it consistent with rest of the file.

**Review by batunpc — CHANGES_REQUESTED** [review]
Thank you for this PR @devils2ndself 

I have left minor feedback regarding Eslint. 

Some parts of the code using `''` and some `""` .  I have pointed out the first one but there are other parts where `''` is used. 

Please replace them with `""`

**Review by batunpc — APPROVED** [review]

**Pull request merged**
